pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -260,20 +266,26 @@ func (pl *PodTopologySpread) calPreFilterState(ctx context.Context, pod *v1.Pod)
 			klog.ErrorS(nil, "Node not found")
 			return
 		}
-		// In accordance to design, if NodeAffinity or NodeSelector is defined,
-		// spreading is applied to nodes that pass those filters.
-		// Ignore parsing errors for backwards compatibility.
-		match, _ := requiredSchedulingTerm.Match(node)
-		if !match {
-			return
-		}
+
 		// Ensure current node's labels contains all topologyKeys in 'Constraints'.
 		if !nodeLabelsMatchSpreadConstraints(node.Labels, constraints) {
 			return
 		}
 
 		tpCounts := make(map[topologyPair]int, len(constraints))
 		for _, c := range constraints {
+			if c.NodeAffinityPolicy == honor {
+				if match, _ := requiredNodeAffinity.Match(node); !match {
+					continue
```

Comment:
We should `return`, same for the NodeTaintsPolicy check below, as well as the logic in score.

Reply:
Why should we directly return. By default, it behaves the same with current implement. When we specified multi constraints, we should still respect others constraints in calculating.

Reply:
Agree. In the past, the implicit policy applied to the node. Now it is per-constraint.

Reply:
You're right. I was precepted that Node*Policy in multiple constraints always hold the same honor or ignore value.

Can you add UTs to cover this case:

- constraint1: topologyKey=node, NodeAffinityPolicy=Honor
- constarint2: topologyKey=zone, NodeAffinityPolicy=Ignore

Other than this, integration test are still a must.

Reply:
done :)